### ISSUE TITLE
How to use methodThrowTypeDeclarations

### ISSUE THREAD

**Issue by Stadly** [pepakriz/phpstan-exception-rules#22]
I cannot get `methodThrowTypeDeclarations` to work. Consider the following code:

```php
<?php

use GuzzleHttp\Psr7\LimitStream;
use Psr\Http\Message\StreamInterface;

class Foo {
    /** @var StreamInterface */
    private $stream;

    public function __construct(StreamInterface $stream, int $limit, int $offset) {
        try {
            $this->stream = new LimitStream($stream, $limit, $offset);
        } catch (RuntimeException $exception) {
            // Do something
        }
    }
}
```

Using a configuration like this:

```yaml
parameters:
    exceptionRules:
        checkedExceptions:
            - RuntimeException
        methodThrowTypeDeclarations:
            GuzzleHttp\Psr7\LimitStream:
                __construct:
                    - RuntimeException

includes:
    - vendor/pepakriz/phpstan-exception-rules/extension.neon
```

I get the following error report:

```
 ------ -----------------------------------------------------------------
  Line   src\test.php
 ------ -----------------------------------------------------------------
  13     RuntimeException is never thrown in the corresponding try block
 ------ -----------------------------------------------------------------
```

What am I doing wrong?

**Comment by pepakriz** [comment]
Hi @Stadly . Currently, constructors are not supported in `methodThrowTypeDeclarations` and in `throw type extensions`. But this feature is planned for next release.

**Comment by pepakriz** [comment]
Solved by 60b1c46f0a5ddbdf16706b34bd8c669dee867c2e and 531c9fe76b2e9ac6adab7615a217a440816ef57c

And released as https://github.com/pepakriz/phpstan-exception-rules/releases/tag/v0.3.1
